reference/sifive-hifive-unmatched.rst

```diff
@@ -2,18 +2,178 @@
+
 
 Using the pre-installed server image
 ====================================
 
+#. Flash the pre-installed server image to a microSD card (see
+   :doc:`/how-to/flash-images`)
+
+#. Insert the microSD card into the board
+
+#. Set the boot source to the microSD card (see `Boot source selection`_)
+
+#. Connect the USB UART cable to the :term:`UART` header (see `UART console`_
+   and :doc:`/how-to/uart-console`)
+
+#. Power on the board
+
+#. Wait for an output line confirming that `cloud-init`_ has finished running;
+   this service is responsible for generating SSH keys, and creating the
+   default user:
+
+   .. code-block:: text
+
+       [   35.682018] cloud-init[909]: Cloud-init v. 24.1.3-0ubuntu3 finished at Tue, 23 Apr 2024 07:44:59 +0000. Datasource DataSourceNoCloud [seed=/var/lib/cloud/seed/nocloud-net][dsmode=net].  Up 35.65 seconds
+
+#. Login with the user *ubuntu* and the default password *ubuntu*; you will be
+   asked to choose a new password
+
 
 Using the live server image
 ===========================
 
+The live installer image is used to install Ubuntu to an :term:`NVMe` drive
+attached to the M.2 slot on the board.
+
+#. Flash the pre-installed server image to a microSD card (see
+   :doc:`/how-to/flash-images`)
+
+#. Insert the microSD card into the board
+
+#. Set the boot source to the microSD card (see `Boot source selection`_)
+
+#. Connect the USB UART cable to the :term:`UART` header (see `UART console`_
+   and :doc:`/how-to/uart-console`)
+
+#. Power on the board
+
+#. When "Hit any key to stop autoboot" is displayed, press :kbd:`Enter`
+
+#. Enter the following commands to boot the installer:
+
+   .. code-block:: text
+
+       pci enum
+       nvme scan
+       load mmc 0:1 $fdt_addr_r dtb/sifive/hifive-unmatched-a00.dtb
+       load mmc 0:1 $kernel_addr_r EFI/boot/bootriscv64.efi
+       bootefi $kernel_addr_r $fdt_addr_r
+
+#. From the GRUB menu, select "Try or Install Ubuntu Server"
+
+#. Loading the installer takes some time. Once it is loaded, follow the
+   `Ubuntu Server installation tutorial
+   <https://ubuntu.com/tutorials/install-ubuntu-server>`_
+
+#. Once the installation is complete, reboot the board without removing the
+   microSD card
+
+#. Press :kbd:`Enter` when "Hit any key to stop autoboot" is displayed, and
+   enter:
+
+   .. code-block:: text
+
+       pci enum
+       nvme scan
+       efidebug boot add -b 0001 'Ubuntu' nvme 0:1 /EFI/ubuntu/grubriscv64.efi
+       efidebug boot order 0001
+       bootefi bootmgr
+
+#. On the next boot, U-Boot will automatically start GRUB
+
+.. note::
+
+    U-Boot does not allow the operating system to write :term:`UEFI` variables.
+    You can do this manually using U-Boot's :command:`eficonfig` command.
+
+
+Boot source selection
+=====================
+
+The SiFive HiFive Unmatched board can boot firmware from the built-in flash
```

Comment:
In our Wiki we clear said use the 1011 setting. We never tested if 1000 would boot with our SD card.

If even the silk screen is incorrect on some boards it is getting even more confusing.

The appended picture might be more helpful to describe which setting we need.

![SiFive_Unmatched_Boot_Mode_Switches](https://github.com/canonical/ubuntu-boards-documentation/assets/2747104/1ce89312-fbca-4e33-9370-4deba91b1a74)

Reply:
Pushed a couple of commits that should make this clearer